test/testframe.c

```diff
@@ -391,140 +530,77 @@ TestShutdown(void)
 H5_ATTR_PURE int
 GetTestVerbosity(void)
 {
-    return (TestVerbosity);
+    return TestVerbosity_g;
 }
 
 /*
- * Set the verbosity level for the testing framework.
- * Return previous verbosity level.
+ * Set the verbosity level for the testing framework
  */
 int
 SetTestVerbosity(int newval)
 {
     int oldval;
 
-    oldval        = TestVerbosity;
-    TestVerbosity = newval;
-    return (oldval);
+    if (newval < 0)
+        newval = VERBO_NONE;
+    else if (newval > VERBO_HI)
+        newval = VERBO_HI;
+
+    oldval          = TestVerbosity_g;
+    TestVerbosity_g = newval;
+
+    return oldval;
 }
 
 /*
  * Retrieve the TestExpress mode for the testing framework
- Values:
- 0: Exhaustive run
-    Tests should take as long as necessary
- 1: Full run.  Default if H5_TEST_EXPRESS_LEVEL_DEFAULT
-    and HDF5TestExpress are not defined
-    Tests should take no more than 30 minutes
- 2: Quick run
-    Tests should take no more than 10 minutes
- 3: Smoke test.
-    Default if HDF5TestExpress is set to a value other than 0-3
-    Tests should take less than 1 minute
-
- Design:
- If the environment variable $HDF5TestExpress is defined,
- or if a default testing level > 1 has been set via
- H5_TEST_EXPRESS_LEVEL_DEFAULT, then test programs should
- skip some tests so that they
- complete sooner.
-
- Terms:
- A "test" is a single executable, even if it contains multiple
- sub-tests.
- The standard system for test times is a Linux machine running in
- NFS space (to catch tests that involve a great deal of disk I/O).
-
- Implementation:
- I think this can be easily implemented in the test library (libh5test.a)
- so that all tests can just call it to check the status of $HDF5TestExpress.
  */
 int
 GetTestExpress(void)
 {
-    char *env_val;
-
-    /* set it here for now.  Should be done in something like h5test_init(). */
-    if (TestExpress == -1) {
-        int express_val = 1;
-
-        /* Check if a default test express level is defined (e.g., by build system) */
-#ifdef H5_TEST_EXPRESS_LEVEL_DEFAULT
-        express_val = H5_TEST_EXPRESS_LEVEL_DEFAULT;
-#endif
-
-        /* Check if HDF5TestExpress is set to override the default level */
-        env_val = getenv("HDF5TestExpress");
-        if (env_val) {
-            if (strcmp(env_val, "0") == 0)
-                express_val = 0;
-            else if (strcmp(env_val, "1") == 0)
-                express_val = 1;
-            else if (strcmp(env_val, "2") == 0)
-                express_val = 2;
-            else
-                express_val = 3;
-        }
-
-        SetTestExpress(express_val);
-    }
-
-    return (TestExpress);
+    return h5_get_testexpress();
```

Comment:
Uses the functionality from h5test.c. The goal is to make it easy for testframe.c tests to use functionality from both frameworks if need be, while making it more difficult for h5test.c-style tests to use functionality from testframe.c.